tests/rptest/services/action_injector.py

```diff
@@ -0,0 +1,183 @@
+        raise NotImplementedError
+
+    def __call__(self, *args, **kwargs):
+        if not self.limit_reached():
+            self.action()
+
+
+class RandomNodeDecommission(RandomNodeOp):
+    def limit_reached(self):
+        return len(self.nodes_affected) >= self.config.max_affected_nodes
+
+    def action(self):
+        node = self.target_node()
+        if node:
+            broker_id = next((i for i, n in enumerate(self.redpanda.nodes) if n == node))
+            self.redpanda.stop_node(node)
+            self.admin.decommission_broker(id=broker_id)
+            wait_until(
+                lambda: broker_id not in {b['node_id'] for b in self.admin.get_brokers()},
+                timeout_sec=120,
+                backoff_sec=2,
+                err_msg=f'Failed to decommission broker id {broker_id}'
+            )
+            self.nodes_affected.add(node)
+
+
+class RandomLeadershipTransfer(RandomNodeOp):
+
+    def __init__(
+            self,
+            redpanda: RedpandaService,
+            config: ActionConfig,
+            admin: Admin,
+            topics: List[TopicSpec],
+    ):
+        super().__init__(redpanda, config, admin)
+        self.topics = topics
+
+    def limit_reached(self):
+        return False
+
+    def action(self):
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                old_leader = self.admin.get_partition_leader(namespace='kafka', topic=topic, partition=partition)
+                self.admin.transfer_leadership_to(namespace='kafka',
+                                                  topic=topic,
+                                                  partition=partition,
+                                                  target=None)
+
+                def leader_is_changed():
+                    new_leader = self.admin.get_partition_leader(namespace='kafka', topic=topic, partition=partition)
+                    return new_leader != -1 and new_leader != old_leader
+
+                wait_until(leader_is_changed,
+                           timeout_sec=30,
+                           backoff_sec=2,
+                           err_msg='Leadership transfer failed')
+
+
+class RandomNodeProcessFailure(RandomNodeOp):
+    def __init__(self, redpanda: RedpandaService, config: ActionConfig, admin: Admin):
+        super(RandomNodeProcessFailure, self).__init__(redpanda, config, admin)
+        self.failure_injector = FailureInjector(self.redpanda)
+
+    def limit_reached(self):
+        return len(self.nodes_affected) >= self.config.max_affected_nodes
+
+    def action(self):
+        node = self.target_node()
+        if node:
+            self.failure_injector.inject_failure(FailureSpec(FailureSpec.FAILURE_KILL, node))
+            self.nodes_affected.add(node)
+
+
+class ActionInjectorThread(Thread):
+    def __init__(
+            self,
+            config: ActionConfig,
+            redpanda: RedpandaService,
+            random_op: RandomNodeOp,
```

Comment:
`random_op` is a misleading name

Reply:
I have changed the name to `disruptive_action`, let me know your thoughts on this.